### PR TITLE
Add styling and mustache rendering and sorting and filtering features

### PR DESCRIPTION
**Pull request by rmr167**
The main files are: forum.html, forum-script.js, ForumElement.java, ForumServlet.java, index.yaml, and forum-style.css.  The general.css was pulled from tutorials branch and may change. The main features added are the mustache rending, the sorting and filtering of the questions on the forum, and the basic styling.

### CODE REVIEW

**Comment by nihabhaskar** [comment]
Looks good, sorry took me a while to understand what was going on. I don't have any comments on the styling cause I have not seen exactly what it looks like & I know you were planning on making some styling changes.

File: src/main/webapp/scripts/forum-template.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const forumTemplate = '<div class="forum-element {{elementType}}">\n' +
```
Comment (geoffreyzgoog) [inline src/main/webapp/scripts/forum-template.js:1]: Is it possible to put the template in an HTML file? That would make formatting and reviewing much easier.

Comment (rmr167) [reply src/main/webapp/scripts/forum-template.js:1]: We originally had it in the html file in a script tag but the linter gave us errors.  Should I move it back and proceed with errors?

File: src/main/webapp/forum.html [hunk 2]
```diff
@@ -3,49 +3,96 @@
...
         </div>
         <div id="new-question-logged-out">
-          <h1>Want to Create a Question?</h1>
+          <h2>Want to Create a Question?</h2>
           <form action="login.html">
             <!-- TODO: Need to get login url -->
             <input type="submit" value="Login" />
           </form>
-          <h1>Don't have an Account?</h1>
+          <h2>Don't have an Account?</h2>
```
[the remark this answers is not in the document]
Comment (rmr167) [reply src/main/webapp/forum.html]: Changed to: Don't have an account?

File: src/main/webapp/forum.html [hunk 3]
```diff
@@ -3,49 +3,96 @@
...
-            <input type="submit" value="Post">
+            </select><br/>
+            <label for="text-input">Question:</label>
+            <input type="text" id="text-input" name="text"><br/>
+            <input id="primary-button" type="submit" value="Post">
           </form>
         </div>
         <div id="new-question-logged-out">
-          <h1>Want to Create a Question?</h1>
+          <h2>Want to Create a Question?</h2>
```
[the remark this answers is not in the document]
Comment (rmr167) [reply src/main/webapp/forum.html]: Changed to: Want to create a question?

File: src/main/webapp/scripts/forum-template.js (hunk 1 above)
Comment (geoffreyzgoog) [reply src/main/webapp/scripts/forum-template.js:1]: It's fine, Niha told me about the linter issue. Can you add a comment in this file with a note on that? I.e. why it's in JS but not in HTML.

Comment (rmr167) [reply src/main/webapp/scripts/forum-template.js:4]: Added comment to javascript file.

**Pull request merged**